src/react/Reconciler.js

```diff
@@ -0,0 +1,315 @@
+  // extract an execution error from the Prepack environment.
+  let context = new ExecutionContext();
+  realm.pushContext(context);
+  try {
+    let message = (Get(realm, (completionValue: any).value, "message"): any).value;
+    let stack = (Get(realm, (completionValue: any).value, "stack"): any).value;
+    let error = new Error("Error evaluating function");
+    error.stack = message + "\n" + stack;
+    return error;
+  } finally {
+    realm.popContext(context);
+  }
+}
+
+function createObject(realm: Realm, shape: null | { [id: string]: any }, name: string | null) {
+  let obj = ObjectCreate(realm, realm.intrinsics.ObjectPrototype);
+  if (shape != null) {
+    // to get around Flow complaining that shape could be null
+    let shapeThatIsNotNull = shape;
+    Object.keys(shape).forEach((id: string) => {
+      let value = shapeThatIsNotNull[id];
+      obj.$Set(id, value, obj);
+      if (name !== null) {
+        value.intrinsicName = `${name}.${id}`;
+      }
+    });
+  }
+  if (name !== null) {
+    obj.intrinsicName = name;
+  }
+  return obj;
+}
+
+function createAbstractByType(realm: Realm, typeNameString: any, name: string) {
+  let type = Value.getTypeFromName(typeNameString);
+  invariant(type !== undefined, "createAbstractByType() cannot be undefined");
+  let value = AbstractValue.createFromTemplate(realm, buildExpressionTemplate(name), type, [], name);
+  value.intrinsicName = name;
+  return value;
+}
+
+function _createAbstractObject(realm: Realm, name: string, properties): AbstractValue {
+  let value = AbstractValue.createFromTemplate(realm, buildExpressionTemplate(name), ObjectValue, [], name);
+  value.intrinsicName = name;
+  let template = createObject(realm, properties, name);
+  template.makePartial();
+  template.makeSimple();
+  value.values = new ValuesDomain(new Set([template]));
+  realm.rebuildNestedProperties(value, name);
+  return value;
+}
+
+function createAbstractObject(realm: Realm, name: string | null, propTypes: any): ObjectValue | AbstractValue {
+  if (propTypes !== null) {
+    let propTypeObject = {};
+
+    Object.keys(propTypes).forEach(key => {
+      let value = propTypes[key];
+      let propertyName = name !== null ? `${name}.${key}` : key;
+      if (typeof value === "string") {
+        propTypeObject[key] = createAbstractByType(realm, value, propertyName);
+      } else if (typeof value === "object" && value !== null) {
+        propTypeObject[key] = createAbstractObject(realm, propertyName, value);
+      } else {
+        invariant(false, `Unknown propType value of "${value}" for "${key}"`);
+      }
+    });
+
+    return _createAbstractObject(realm, name || "unknown", propTypeObject);
+  } else {
+    return _createAbstractObject(realm, name || "unknown", null);
+  }
+}
+
+function callFunction(realm: Realm, functionValue: FunctionValue, thisArg, argsList = []) {
+  if (thisArg === undefined) {
+    thisArg = realm.intrinsics.undefined;
+  }
+  let context = new ExecutionContext();
```

Comment:
This setup should be done by the $Call function. The only work that should be necessary is things mirroring `Call` in `src/methods/call.js`